### DISCUSSION TITLE
"Missing defaultValue" warning with AntDesign 3.x Select + placeholder

### DISCUSSION THREAD

**Discussion by lucksp** [react-hook-form/react-hook-form#3709] [General] [answered]
I have tried researching a bunch - including https://github.com/react-hook-form/react-hook-form/discussions/2656 and https://spectrum.chat/react-hook-form/help/another-problem-with-usefieldarray-swap-and-controller~0a9dfaad-1b0b-49d7-84e6-0164fe35950d without success.

Our issue is that using React-Hook-Form's `<Controller />` around AntDesign's 3.x `<Select />` using a `placeholder` for the Dropdown, does not render the dropdown.  It for some reason is `display: none`.  The apparent fix to display the placeholder is to define a width.  However, this still does not work if the Form's `defaultValue` is an empty string `''`.  Only by setting the value to `undefined` does the placeholder appear as expected...except for the React-Hook-Form warnings.

Here is a codesandbox: https://codesandbox.io/s/throbbing-tree-7lsd5?file=/src/App.js

As you can see, the `undefined` default allows the placeholder to be visible...but not without the warning
![image](https://user-images.githubusercontent.com/15162169/102411907-f78d6e80-3faf-11eb-8ce0-afc6bf64005f.png)

I have tried to `register()` but no success either.

Any ideas what needs to be updated?

**Comment by bluebill1049** [comment, marked as answer]
Try this: https://codesandbox.io/s/happy-ives-15nxv?file=/src/App.js

**Reply by lucksp** [reply]
Wow, awesome.  THANK YOU.  I missed the render prop as an option.  Safe to mark sovled.

**Reply by bluebill1049** [reply]
no worries 👍
